src/components/views/elements/ImageView.tsx

```diff
@@ -309,15 +312,6 @@ export default class ImageView extends React.Component<IProps, IState> {
         this.setZoomAndRotation(cur + 90);
     };
 
-    private onDownloadClick = (): void => {
-        const a = document.createElement("a");
-        a.href = this.props.src;
-        if (this.props.name) a.download = this.props.name;
```

Comment:
Oh, so that's why the download button was just opening (some) files in a new tab. It was the files with no name.